addons/binding/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/handler/core/TelldusCoreDeviceController.java

```diff
@@ -0,0 +1,348 @@
+            throws TellstickException {
+
+        if (!workerThread.isAlive()) {
+            workerThread.start();
+        }
+
+        Long eventTime = System.currentTimeMillis();
+        synchronized (messageQue) {
+            messageQue.put(device, new TelldusCoreSendEvent(device, resendCount, isDimmer, command, eventTime));
+            messageQue.notify();
+        }
+    }
+
+    @Override
+    public State calcState(Device dev) {
+        TellstickDevice device = (TellstickDevice) dev;
+        State st = null;
+        switch (device.getStatus()) {
+            case JNA.CLibrary.TELLSTICK_TURNON:
+                st = OnOffType.ON;
+                break;
+            case JNA.CLibrary.TELLSTICK_TURNOFF:
+                st = OnOffType.OFF;
+                break;
+            case JNA.CLibrary.TELLSTICK_DIM:
+                BigDecimal dimValue = new BigDecimal(device.getData());
+                if (dimValue.intValue() == 0) {
+                    st = OnOffType.OFF;
+                } else if (dimValue.intValue() >= 255) {
+                    st = OnOffType.ON;
+                } else {
+                    st = OnOffType.ON;
+                }
+                break;
+            default:
+                logger.warn("Could not handle {} for {}", device.getStatus(), device);
+        }
+        return st;
+    }
+
+    @Override
+    public BigDecimal calcDimValue(Device device) {
+        BigDecimal dimValue = new BigDecimal(0);
+        switch (((TellstickDevice) device).getStatus()) {
+            case JNA.CLibrary.TELLSTICK_TURNON:
+                dimValue = new BigDecimal(100);
+                break;
+            case JNA.CLibrary.TELLSTICK_TURNOFF:
+                break;
+            case JNA.CLibrary.TELLSTICK_DIM:
+                dimValue = new BigDecimal(((TellstickDevice) device).getData());
+                dimValue = dimValue.multiply(new BigDecimal(100));
+                dimValue = dimValue.divide(new BigDecimal(255), 0, BigDecimal.ROUND_HALF_UP);
+                break;
+            default:
+                logger.warn("Could not handle {} for {}", ((TellstickDevice) device).getStatus(), device);
+        }
+        return dimValue;
+    }
+
+    public long getLastSend() {
+        return lastSend;
+    }
+
+    public void setLastSend(long currentTimeMillis) {
+        lastSend = currentTimeMillis;
+    }
+
+    @Override
+    public void onRequest(TellstickSensorEvent newDevices) {
+        setLastSend(newDevices.getTimestamp());
+    }
+
+    @Override
+    public void onRequest(TellstickDeviceEvent newDevices) {
+        setLastSend(newDevices.getTimestamp());
+
+    }
+
+    private void sendEvent(Device device, int resendCount, boolean isdimmer, Command command)
+            throws TellstickException {
+        for (int i = 0; i < resendCount; i++) {
+            checkLastAndWait(resendInterval);
+            logger.info("Send " + command + " to " + device + " times=" + i);

```

Comment:
reduce to debug and change to parameterized logging instead of string concatenation